src/core/public/chrome/ui/header/nav_link.tsx

```diff
@@ -71,6 +74,29 @@ export function createEuiListItem({
   };
 }
 
+export function createEuiButtonItem({
+  link,
+  onClick = () => {},
+  navigateToUrl,
+  dataTestSubj,
+}: Omit<Props, 'appId' | 'basePath'>) {
```

Comment:
The reason I used `omit` is because it is essentially creating the same link item just in a different format. The format being an EuiButton that doesn't have a `isActive` nor do we want a custom icon.